session/session.go

```diff
@@ -0,0 +1,34 @@
+package session
+
+import (
+	"fmt"
+	"time"
+
+	"launchpad.net/gozk"
+)
+
+var ErrNotConnected = fmt.Errorf("Unable to connect to ZooKeeper.")
+
+type ZkSession struct {
+	Connection *zookeeper.Conn
+	Events     <-chan zookeeper.Event
+}
+
+func NewZkSession(servers string) (*ZkSession, error) {
+	zkconn, zkevents, err := zookeeper.Dial(servers, 3*time.Second)
+
+	if err != nil {
+		return nil, err
+	}
+
+	select {
+	case event := <-zkevents:
+		if event.State != zookeeper.STATE_CONNECTED {
+			return nil, ErrNotConnected
+		}
+
+		return &ZkSession{zkconn, zkevents}, nil
+	case <-time.After(3 * time.Second):
```

Comment:
> I think this will already happen courtesy of gozk after 3 seconds, but I'm not positive.

Unfortunately, it doesn't. I tested before adding this. The timeout passed to `Dial()` is defined as:

*The recvTimeout parameter, given in nanoseconds, allows controlling the amount of time the connection can stay unresponsive before the server will be considered problematic.*

It turns out that it only applies after you get a successful connection. The effect of this timeout value is pretty subtle. I need to do some more investigation.